resources/js/Pages/Articles/Components/LatestArticles.tsx

```diff
@@ -0,0 +1,78 @@
+import cn from "classnames";
+import { ArticleCard } from "@/Components/Articles/ArticleCard";
+import { ArticleCardSkeleton } from "@/Components/Articles/ArticleCard/ArticleCardSkeleton";
+import { Carousel, CarouselItem } from "@/Components/Carousel";
+import { useBreakpoint } from "@/Hooks/useBreakpoint";
+
+export const LatestArticles = ({
+    articles,
+    withFullBorder,
+    isLoading,
+    hasEnoughArticles,
+}: {
+    articles: App.Data.Articles.ArticleData[];
+    withFullBorder: boolean;
+    isLoading: boolean;
+    hasEnoughArticles: boolean;
+}): JSX.Element => {
+    const slidesPerView = useArticlesCarousel();
+
+    const pagination = {
+        clickable: true,
+        renderBullet: (_index: number, className: string) => `<span class="h-1.5 rounded-xl w-6 ${className}"></span>`,
+    };
+
+    const articlePlaceHolders = Array.from({ length: 3 - articles.length });
+
+    return (
+        <div className={cn("w-full pt-2 ", { "mb-6": !withFullBorder, "lg:mb-6": withFullBorder })}>
+            <div
+                className={cn("latest-articles-carousel w-full", {
+                    "pb-6": slidesPerView < 3,
+                })}
+            >
+                <Carousel
+                    swiperClassName="flex pb-6"
+                    shouldShowHeader={false}
+                    slidesPerView={slidesPerView}
+                    pagination={pagination}
+                >
+                    {articles.map((article, index) => (
+                        <CarouselItem key={index}>
+                            <ArticleCard
+                                article={article}
+                                variant="large"
+                            />
+                        </CarouselItem>
+                    ))}
+
+                    {articlePlaceHolders.map((_v, index) => (
+                        <CarouselItem key={index}>
+                            <ArticleCardSkeleton
+                                key={index}
+                                withImgPlaceholder={isLoading}
+                            />
+                        </CarouselItem>
+                    ))}
+                </Carousel>
+            </div>
+            <div
+                className={cn("mt-0", {
+                    "-mx-6 -mt-6 space-y-2 border-b-4 border-theme-secondary-100 sm:-mx-8 lg:mx-0 lg:border-b-2 lg:border-theme-secondary-300":
+                        withFullBorder && (hasEnoughArticles || isLoading),
+                    "border-b-2 border-theme-secondary-300": !withFullBorder && (hasEnoughArticles || isLoading),
+                })}
+            ></div>
+        </div>
+    );
+};
+
+const useArticlesCarousel = (): number => {
```

Comment:
this name is odd as it returns a number of slides to use 🤔 based on the name I had expected something else like a boolean that determines whether it uses a carousel

Reply:
Should be fine